src/state/slices/portfolioSlice/utils.ts

```diff
@@ -270,13 +227,51 @@ export const accountToPortfolio: AccountToPortfolio = ({
         break
       }
       default:
-        break
+        assertUnreachable(chainNamespace)
     }
   })
 
   return portfolio
 }
 
+export const checkAccountHasActivity = (account: Account<ChainId>) => {
+  const { chainId } = account
+  const { chainNamespace } = fromChainId(chainId)
+
+  switch (chainNamespace) {
+    case CHAIN_NAMESPACE.Evm: {
+      const ethAccount = account as Account<KnownChainIds.EthereumMainnet>
+
+      const hasActivity =
+        bnOrZero(ethAccount.chainSpecific.nonce).gt(0) || bnOrZero(ethAccount.balance).gt(0)
+
+      return hasActivity
+    }
+    case CHAIN_NAMESPACE.Utxo: {
+      const utxoAccount = account as Account<UtxoChainId>
+      const { balance } = account
+
+      const hasActivity =
+        bnOrZero(balance).gt(0) ||
+        bnOrZero(utxoAccount.chainSpecific.nextChangeAddressIndex).gt(0) ||
+        bnOrZero(utxoAccount.chainSpecific.nextReceiveAddressIndex).gt(0)
+
+      return hasActivity
+    }
+    case CHAIN_NAMESPACE.CosmosSdk: {
+      const cosmosAccount = account as Account<KnownChainIds.CosmosMainnet>
+
+      const hasActivity =
+        bnOrZero(cosmosAccount.chainSpecific.sequence).gt(0) ||
```

Comment:
Ditto here balance checks most likely not needed, but absolutely doesn't hurt to!

Reply:
Yeah this guy's lift and shift so logic is preserved from previous implementation